ddtrace/tracer/option.go

```diff
@@ -243,12 +243,18 @@ func WithSampler(s Sampler) StartOption {
 	}
 }
 
-// WithHTTPRoundTripper allows customizing the underlying HTTP transport for
-// emitting spans. This is useful for advanced customization such as emitting
-// spans to a unix domain socket. The default should be used in most cases.
+// WithHTTPRoundTripper is deprecated. Please consider using WithHTTPClient instead.
```

Comment:
Do we want to leave some of the original doc here alongside the deprecation warning?

I don't like when libraries remove older documentation because it makes it more difficult to figure out what an application using the library is *trying* to do. If the deprecated documentation is still in place, it's easier for me to tell without reading the library source what's going on.

I can see that leaving the documentation here might be seen by users as an endorsement for using the function. Maybe we can remove the `This is useful for` part, since that doesn't clarify what it does, and suggests its use.

```suggestion
// WithHTTPRoundTripper is deprecated. Please consider using WithHTTPClient instead.
// The function allows customizing the underlying HTTP transport for emitting spans. 
```

Reply:
Makes perfect sense.